lib/rules/prefer-object-spread.js

```diff
@@ -0,0 +1,249 @@
+         */
+        const formattedObjectLiteral = text.replace(/^(.*){ */, "$1").replace(/ *}([^}]*)$/, "$1");
+
+        return isLast ? formattedObjectLiteral : addComma(formattedObjectLiteral, comments);
+    }
+
+    if (isLast) {
+        return parens ? `${spread}(${text})` : `${spread}${text}`;
+    }
+
+    return parens ? addComma(`${spread}(${text})`, comments) : `${spread}${addComma(text, comments)}`;
+}
+
+/**
+ * Autofixes the Object.assign call to use an object spread instead.
+ * @param {ASTNode|null} node - The node that the rule warns on, i.e. the Object.assign call
+ * @param {string} sourceCode - sourceCode of the Object.assign call
+ * @returns {Function} autofixer - replaces the Object.assign with a spread object.
+ */
+function autofixSpread(node, sourceCode) {
+    return fixer => {
+        const args = node.arguments;
+        const firstArg = args[0];
+        const lastArg = args[args.length - 1];
+        const parens = needsParens(node);
+        const comments = sourceCode.getCommentsInside(node);
+        const replaceObjectAssignStart = fixer.replaceTextRange(
+            [node.range[0], firstArg.range[0]],
+            `${parens ? "({" : "{"}`
+        );
+
+        const handleArgs = args
+            .map((arg, i, arr) => formatArg(arg, i + 1 >= arr.length, sourceCode, comments))
+            .filter(arg => arg !== "," && arg !== "");
+
+        const insertBody = fixer.replaceTextRange([firstArg.range[0], lastArg.range[1]], handleArgs.join(""));
+        const replaceObjectAssignEnd = fixer.replaceTextRange([lastArg.range[1], node.range[1]], `${parens ? "})" : "}"}`);
+
+        return [
+            replaceObjectAssignStart,
+            insertBody,
+            replaceObjectAssignEnd
+        ];
+    };
+}
+
+/**
+ * Autofixes the Object.assign call with a single object literal as an argument
+ * @param {ASTNode|null} node - The node that the rule warns on, i.e. the Object.assign call
+ * @param {string} sourceCode - sourceCode of the Object.assign call
+ * @returns {Function} autofixer - replaces the Object.assign with a object literal.
+ */
+function autofixObjectLiteral(node, sourceCode) {
+    return fixer => {
+        const argument = node.arguments[0];
+        const parens = needsParens(node);
+
+        return fixer.replaceText(node, `${parens ? "(" : ""}${sourceCode.text.slice(argument.range[0], argument.range[1])}${parens ? ")" : ""}`);
+    };
+}
+
+
+module.exports = {
+    meta: {
+        docs: {
+            description:
+                "disallow using Object.assign with an object literal as the first argument and prefer the use of object spread instead.",
+            category: "Stylistic Issues",
+            recommended: false,
+            url: "https://eslint.org/docs/rules/prefer-object-spread"
+        },
+        schema: [],
+        fixable: "code",
+        messages: {
+            useSpreadMessage: "Use an object spread instead of `Object.assign()` eg: `{ ...foo }`",
```

Comment:
I don't think the example should be necessary. Users can view the documentation (or search the Internet) for what object spread means.

Reply:
Well, turning that around, why doesn't the object literal case have an example?

Also: Since we have autofix here, the user could just let that be run and see how the item should look. (And no, we don't need to note that `--fix` could be run in the lint message, because we note that in most formatters outside of the lint message)

Reply:
I suppose that's a fair point - maybe it should have an example :-)

(also since autofix can't be easily run for a single rule, that's not really a practical option for demonstrating the results of a single rule)